pkg/icingadb/sla.go

```diff
@@ -0,0 +1,137 @@
+package icingadb
+
+import (
+	"context"
+	"github.com/icinga/icingadb/pkg/contracts"
+	v1 "github.com/icinga/icingadb/pkg/icingadb/v1"
+	"github.com/icinga/icingadb/pkg/types"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+	"time"
+)
+
+type SlaHistoryTrail struct {
+	Id                 types.Int `json:"id" db:"-"`
+	v1.EnvironmentMeta `json:",inline"`
+	HostId             types.Binary    `json:"host_id"`
+	ServiceId          types.Binary    `json:"service_id"`
+	EventType          string          `json:"event_type"`
+	EventTime          types.UnixMilli `json:"event_time"`
+}
+
+// Fingerprint implements the contracts.Fingerprinter interface.
+func (sht SlaHistoryTrail) Fingerprint() contracts.Fingerprinter {
+	return sht
+}
+
+// ID implements part of the contracts.IDer interface.
+func (sht SlaHistoryTrail) ID() contracts.ID {
+	return sht.Id
+}
+
+// SetID implements part of the contracts.IDer interface.
+func (sht *SlaHistoryTrail) SetID(id contracts.ID) {
+	sht.Id = id.(types.Int)
+}
```

Comment:
Just to implement these two methods? Then what about making them no-ops instead? Just an idea...

Reply:
If these methods would be no-ops, you could drop the methods https://github.com/Icinga/icingadb/pull/566#discussion_r1124277436 altogether along with the column https://github.com/Icinga/icingadb/pull/566#discussion_r1124277147 IMAO.

Reply:
Shouldn't the struct reflect the database table? You even told me that yesterday.

Reply:
Yes. But in this case I see no sense. See, we even explicitly set db:"-". But on the other hand you’re right. As I said: Just an idea.